source/common/common/trie.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+namespace {
+// Because vector is efficient when removing elements from the back but paths
+// generally have the most significant components at the front, it's useful to
+// reverse the components vector before processing
+template <class T> inline void reverseComponentVector(std::vector<T>& vec) {
+  std::reverse(vec.begin(), vec.end());
```

Comment:
This was based on a previous comment from me of generally adding a comment and I think I mentioned a helper if that would help. I agree the wrapper is a bit excessive, I just care about comment clarity.